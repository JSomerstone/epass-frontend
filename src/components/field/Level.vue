<template>
  <b-field :label="label" :label-position="labelPosition">
    <b-radio-button v-model="level" :native-value="0" v-if="!hideNational">
        National
    </b-radio-button>
    <b-radio-button v-model="level" :native-value="1" >
        1
    </b-radio-button>
    <b-radio-button v-model="level" :native-value="2" >
        2
    </b-radio-button>
    <b-radio-button v-model="level" :native-value="3" >
        3
    </b-radio-button>
  </b-field>
</template>
<script>
export default {
  props: {
    value: { type: Number, required: true },
    label: { type: String, default: "Level" },
    'label-position': { type: String, default: "on-border" },
    'hide-national': { type: Boolean, default: false},
  },
  data() {
    return {
      level: this.value
    }
  },
  watch: {
    level: function(newValue) {
      this.$emit('input', newValue);
    }
  },
}
</script>