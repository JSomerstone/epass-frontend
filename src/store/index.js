import Vue from "vue";
import Vuex from "vuex";

import axios from "axios";
import config from "../config/index";

import tournaments from "./modules/tournaments";
import referees from "./modules/referees";
import countries from "./modules/countries";
import auth from "./modules/auth";

axios.defaults.baseURL = config.backendUrl;
Vue.use(Vuex);

const store = new Vuex.Store({
  modules: {
    tournaments,
    referees,
    countries,
    auth
  }
});

export default store;