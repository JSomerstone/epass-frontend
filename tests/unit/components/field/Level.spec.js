import { mount } from "@vue/test-utils";
import Vuex from "vuex";
import localVue from "../../localVue";
import store from "../../mockStore";

import Level from "@/components/field/Level";

describe("Level", () => {

  let propsData;
  beforeAll(() => {
    propsData = { value: 1 }
  });
  it("Renders correctly", () => {
    const wrapper = mount(Level, {
      localVue,
      propsData,
    });
    expect(wrapper).toMatchSnapshot();
  });

  it("Value change is emitted", async () => {
    const wrapper = mount(Level, {
      localVue,
      store: new Vuex.Store(store),
      propsData,
    });
    
    await wrapper.setData({ level: 2 });
    expect(wrapper.vm.level).toBe(2);
  });

});