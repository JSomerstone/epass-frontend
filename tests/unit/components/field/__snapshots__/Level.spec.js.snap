// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Level Renders correctly 1`] = `
<div class="field is-floating-label has-addons"><label class="label">Level</label>
  <div class="control"><label class="b-radio radio button">
      National
      <input type="radio" value="0"></label></div>
  <div class="control"><label class="b-radio radio button is-primary">
      1
      <input type="radio" value="1"></label></div>
  <div class="control"><label class="b-radio radio button">
      2
      <input type="radio" value="2"></label></div>
  <div class="control"><label class="b-radio radio button">
      3
      <input type="radio" value="3"></label></div>
  <!---->
</div>
`;
